src/pkg/packager/components.go

```diff
@@ -80,27 +93,60 @@ func (p *Packager) getValidComponents() []types.ZarfComponent {
 
 		// If the user has requested a choice group, then prompt them
 		if userChoicePrompt {
-			selectedComponent := p.confirmChoiceGroup(componentGroup)
+			selectedComponent := confirmChoiceGroup(componentGroup)
 			validComponentsList = append(validComponentsList, selectedComponent)
 		}
 	}
 
 	// Ensure all user requested components are valid
-	if err := p.validateRequests(validComponentsList, requestedNames, choiceComponents); err != nil {
+	if err := validateRequests(validComponentsList, requestedComponents, choiceComponents); err != nil {
 		message.Fatalf(err, "Invalid component argument, %s", err)
 	}
 
 	return validComponentsList
 }
 
+func (p *Packager) forRequestedComponents(function func(types.ZarfComponent) error) error {
+	requestedComponents := helpers.StringToSlice(p.cfg.PkgOpts.OptionalComponents)
+	partialMirror := len(requestedComponents) > 0 && requestedComponents[0] != ""
+
+	for _, component := range p.cfg.Pkg.Components {
+		included, excluded := false, false
+
+		if partialMirror {
+			included, excluded = includedOrExcluded(component, requestedComponents)
```

Comment:
if this function is meant to be used more generically, the `partialMirror` var is a confusing name. Perhaps `isPartial`?